File: scripts/Python/02_read_files/read_multiple_files.py
import duckdb
import pandas as pd

# all parquet files
files = "main/**/*.parquet"
# open an in-memory DuckDB connection
con = duckdb.connect()

# Prepare your query to read multiple parquet files
# Use where clause to load only needed data
my_data = con.execute(f"""
    SELECT code, user
    FROM '{files}'
    WHERE code = 100
""").fetchdf()

# close the connection
con.close()
# print value of my_data
print(my_data)

File: scripts/R/02_read_files/read_multiple_files.R
library(duckdb)
library(data.table)

# memory DuckDB connection
duckcon <- dbConnect(duckdb::duckdb())

# Query all Parquet files in subfolders like main/sub/names.parquet
files <- glue::glue("main/**/*.parquet")

# Use where clause to load only needed data
query <- glue::glue("SELECT code, user FROM '{files}' where code = 100")

# Generate a data.table from the query result
my_data <- DBI::dbGetQuery(duckcon, query) |> as.data.table()

# close the connection
dbDisconnect(duckcon, shutdown = TRUE)
# print value of my_data
print(my_data)
